gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/HiveSerDeManager.java

```diff
@@ -79,7 +80,8 @@ protected HiveSerDeManager(State props) {
   public abstract boolean haveSameSchema(HiveRegistrationUnit unit1, HiveRegistrationUnit unit2) throws IOException;
 
   public enum Implementation {
-    AVRO(HiveAvroSerDeManager.class.getName());
+    AVRO(HiveAvroSerDeManager.class.getName()),
```

Comment:
Should we use service providers instead of a hard-coded list of implementations? This prevents third parties from adding their own implementations.

Reply:
Existing code has a fallback to treating the implementation name as a class name and instantiating the class using the name. The implementation names here are aliases to the class name.